Purchases/Networking/Caching/CallbackCache.swift

```diff
@@ -0,0 +1,63 @@
+//
+//  Copyright RevenueCat Inc. All Rights Reserved.
+//
+//  Licensed under the MIT License (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      https://opensource.org/licenses/MIT
+//
+//  CallbackCache.swift
+//
+//  Created by Joshua Liebowitz on 11/18/21.
+
+import Foundation
+
+/**
+ Generic callback cache whose primary usage is to help ensure API calls in flight are not duplicated.
+ Users of this class will store a completion block for any Cacheable API call that is running. If the same request is
+ made while a request is in-flight, the completion block will be added to the list and the API call will not be
+ performed. Once the first API call has finished, the user is required to call `performOnAllItemsAndRemoveFromCache`.
```

Comment:
Adding context on where this came from- this is a refactor of https://github.com/RevenueCat/purchases-ios/pull/867 which comes from https://github.com/RevenueCat/purchases-android/pull/358

I need to dig in more to address your concern. I **_think_** because the POST happens right after a GET, the POST is serial and blocks the next GET from getting queued up (which means it won't be added to the cache).

We **_might_** want to make everything serial

Reply:
I think going serial is best for now 😊

Reply:
I agree.

Reply:
In practice, this won't really happen. 
It does need to be fixed, and I'm working on it, but I'm going to merge since this is the current behavior (tested on main). That way I don't hold up any other work and I can focus on just this fix in a new PR.